pkg/crc/api/handlers.go

```diff
@@ -164,14 +163,16 @@ func (h *Handler) SetConfig(args json.RawMessage) string {
 		}
 		successProps = append(successProps, k)
 	}
-
 	if len(multiError.Errors) != 0 {
-		setConfigResult.Success = false
-		setConfigResult.Error = fmt.Sprintf("%v", multiError)
+		return encodeStructToJSON(client.SetOrUnsetConfigResult{
+			Success: false,
+			Error:   multiError.Error(),
+		})
 	}
```

Comment:
With the early return, the list of keys which were successfully set is no longer returned in the case of an error.